src/simple.test.ts

```diff
@@ -20,27 +23,31 @@ const options = fc.record({
 const tree = fc
   .tuple(leaves, options)
   .chain(([leaves, options]) => fc.tuple(fc.constant(SimpleMerkleTree.of(leaves, options)), fc.constant(options)));
-const treeAndLeaf = fc.tuple(leaves, options).chain(([leaves, options]) =>
+const treeAndLeaf = tree.chain(([tree, options]) =>
   fc.tuple(
-    fc.constant(SimpleMerkleTree.of(leaves, options)),
+    fc.constant(tree),
     fc.constant(options),
-    fc.nat({ max: leaves.length - 1 }).map(index => ({ value: leaves[index]!, index })),
+    fc.nat({ max: tree.length - 1 }).map(index => ({ value: tree.at(index)!, index })),
   ),
 );
-const treeAndLeaves = fc.tuple(leaves, options).chain(([leaves, options]) =>
+const treeAndLeaves = tree.chain(([tree, options]) =>
   fc.tuple(
-    fc.constant(SimpleMerkleTree.of(leaves, options)),
+    fc.constant(tree),
     fc.constant(options),
     fc
-      .uniqueArray(fc.nat({ max: leaves.length - 1 }))
-      .map(indices => indices.map(index => ({ value: leaves[index]!, index }))),
+      .uniqueArray(fc.nat({ max: tree.length - 1 }))
+      .map(indices => indices.map(index => ({ value: tree.at(index)!, index }))),
   ),
 );
 
-fc.configureGlobal({ numRuns: process.env.CI ? 10000 : 100 });
-
 testProp('generates a valid tree', [tree], (t, [tree]) => {
   t.notThrows(() => tree.validate());
+
+  // check leaves enumeration
+  for (const [index, value] of tree.entries()) {
+    t.is(value, tree.at(index)!);
+  }
+  t.is(tree.at(tree.length), undefined);
 });
```

Comment:
This will iterate through all entries of the tree, which are unbounded. We should add a `maxLength` in line 11 and fuzz here:

```suggestion
testProp('generates a valid tree', [treeAndLeaf], (t, [tree, , indices]) => {
  t.notThrows(() => tree.validate());

  // check leaves enumeration
  t.is(indices.value, tree.at(indices.index)!);
  t.is(tree.at(tree.length), undefined);
});
```

Reply:
I don't think the unbounded aspect is an issue here (testing the interration). If we get here, that means the tree construction happened ... which sets the bound.

So yes, I would consider adding a bound in line 11, but I would keep the full `.entries()` check here.

Reply:
Sounds good, thanks for updating!